### PR TITLE
add testing and code coverage to cwms-http-client CWMS-686

### PR DESCRIPTION
**Pull request by adamkorynta**

### CODE REVIEW

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java [hunk 1]
```diff
@@ -42,10 +43,14 @@
     private final Map<String, String> queryParameters = new HashMap<>();
     private final Map<String, String> queryHeaders = new HashMap<>();
 
-    public HttpRequestBuilderImpl(ApiConnectionInfo apiConnectionInfo, String endpoint) {
+    public HttpRequestBuilderImpl(ApiConnectionInfo apiConnectionInfo, String endpoint) throws ServerNotFoundException {
         Objects.requireNonNull(apiConnectionInfo, "API connection info must be defined");
         Objects.requireNonNull(apiConnectionInfo.getApiRoot(), "API root must be defined");
-        this.httpUrl = HttpUrl.parse(apiConnectionInfo.getApiRoot());
+        HttpUrl url = HttpUrl.parse(apiConnectionInfo.getApiRoot());
```
Comment (rma-psmorris) [inline cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:49]: do you know if this supports ipv6?

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
Just one question

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java (hunk 1 above)
Comment (adamkorynta) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java:49]: It does, I added a unit test to demonstrate. It also cleans it up.

**Review by rma-psmorris — APPROVED** [review]
approved

**Pull request merged**